#!/usr/bin/env python
PACKAGE = "pcl_voxel_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("leaf_length", double_t, 0, "Voxel leaf length", 0.1, 0, 10)
gen.add("leaf_width",  double_t, 0, "Voxel leaf width",  0.1, 0, 10)
gen.add("leaf_height", double_t, 0, "Voxel leaf height", 0.1, 0, 10)

gen.add("min_points",  int_t,    0, "Minimum point's per voxel", 0, 0, 100)

gen.add("min_range", double_t,   0, "Minimum distance from lidar for points", 0, 0, 10)

exit(gen.generate(PACKAGE, "pcl_voxel_filter", "Voxel"))